pulpcore/tests/functional/api/using_plugin/test_content_delivery.py

```diff
@@ -102,3 +103,51 @@ def test_remote_artifact_url_update(
     actual_checksum = hashlib.sha256(downloaded_file.body).hexdigest()
     expected_checksum = expected_file_list[0][1]
     assert expected_checksum == actual_checksum
+
+
+@pytest.mark.parallel
+def test_remote_content_changed_with_on_demand(
+    write_3_iso_file_fixture_data_factory,
+    file_repo_with_auto_publish,
+    file_remote_ssl_factory,
+    file_bindings,
+    monitor_task,
+    file_distribution_factory,
+):
+    """When:
+    1. Sync a remote with fileA(digest=123)
+    2. On remote server, change fileA content: fileA(digest=456)
+
+    Then:
+    3. Get fileA from content app will cause a connection-close/incomplete-response.
+    """
```

Comment:
Oh I think the trilogy is GIVEN-WHEN-THEN and fits nicely here:
```suggestion
    """
    GIVEN a remote synced on demand with fileA (digest=123),
    WHEN on the remote server, fileA changed its content (digest=456),
    THEN retrieving fileA from the content app will cause a connection-close/incomplete-response.
    """
```
(The goal is to make it sound like actual human language.)
((The next goal is to make the actual test code read just  like this.))